components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/model/FilterTreeBuilder.java

```
@@ -0,0 +1,257 @@
+            setExpressionNodeValues(filterParts[0], " sw ", filterParts[1], expressionNode);
+        } else if (Pattern.compile(Pattern.quote(" ew "), Pattern.CASE_INSENSITIVE).matcher(filterString).find()) {
+            filterParts = trimmedFilter.split(" ew | EW | eW | Ew ");
+            setExpressionNodeValues(filterParts[0], " ew ", filterParts[1], expressionNode);
+        } else if (Pattern.compile(Pattern.quote(" pr"), Pattern.CASE_INSENSITIVE).matcher(filterString).find()) {
+            //with filter PR, there should not be whitespace after.
+            filterParts = trimmedFilter.split(" pr| PR| pR| Pr");
+            setExpressionNodeValues(filterParts[0], " pr", null, expressionNode);
+        } else if (Pattern.compile(Pattern.quote(" gt "), Pattern.CASE_INSENSITIVE).matcher(filterString).find()) {
+            filterParts = trimmedFilter.split(" gt | GT | gT | Gt ");
+            setExpressionNodeValues(filterParts[0], " gt ", filterParts[1], expressionNode);
+
+        } else if (Pattern.compile(Pattern.quote(" ge "), Pattern.CASE_INSENSITIVE).matcher(filterString).find()) {
+            filterParts = trimmedFilter.split(" ge | GE | gE | Ge ");
+            setExpressionNodeValues(filterParts[0], " ge ", filterParts[1], expressionNode);
+        } else if (Pattern.compile(Pattern.quote(" lt "), Pattern.CASE_INSENSITIVE).matcher(filterString).find()) {
+            filterParts = trimmedFilter.split(" lt | LT | lT | Lt ");
+            setExpressionNodeValues(filterParts[0], " lt ", filterParts[1], expressionNode);
+        } else if (Pattern.compile(Pattern.quote(" le "), Pattern.CASE_INSENSITIVE).matcher(filterString).find()) {
+            filterParts = trimmedFilter.split(" le | LE | lE | Le ");
+            setExpressionNodeValues(filterParts[0], " le ", filterParts[1], expressionNode);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Given filter operator is not supported.");
```

shall we throw an error if the given filter string is invalid/ not supported?